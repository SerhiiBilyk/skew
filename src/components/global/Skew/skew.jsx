import React from 'react';
import ReactDOM from 'react-dom';
import styles from './skew.scss';
import CSSModules from 'react-css-modules';




export const Skew=(props)=>{
  return (
  <div styleName={`skew zindex-${props.index}`}>
    <div styleName='rotate'></div>
  </div>
)};

Skew.defaultProps = {
  index: 0
};

export default CSSModules(Skew,styles, {allowMultiple: true});
