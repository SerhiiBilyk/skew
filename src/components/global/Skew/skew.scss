.skew {
  position: relative;
  height: 0;

}

.rotate {
  width: 100vw;
  transform: rotate(-3.5deg);
  //border: 1px solid red;
  height: 6.45vw;
  top: -3.4vw;
  position: absolute;
  left: -1%;
  background-color: white;
  transform-origin: top;
  //border:1px solid black;
    //overflow-x:hidden;
}

.zindex-0 {
  z-index: 0;
}

.zindex-10 {
  z-index: 10;
}
