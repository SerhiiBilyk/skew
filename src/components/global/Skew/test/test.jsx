import React from 'react';
import Skew from '../skew.jsx';
import  {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

// TODO write more tests, it is created only for folder architecture

/**SNAPSHOT*/
it('<Skew/> renders correctly', () => {
  const tree = renderer.create(
   <Skew />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
