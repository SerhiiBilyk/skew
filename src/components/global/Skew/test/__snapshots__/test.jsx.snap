// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skew/> renders correctly 1`] = `
<div
  className="skew zindex-0"
>
  <div
    className="rotate"
  />
</div>
`;
